matrix-api.c

```diff
@@ -63,27 +55,27 @@ void matrix_api_error(MatrixConnectionData *conn, gpointer user_data,
 void matrix_api_bad_response(MatrixConnectionData *ma, gpointer user_data,
         int http_response_code, JsonNode *json_root)
 {
-    JsonObject *json_obj;
+    JsonObject *json_obj = NULL;
     const gchar *errcode = NULL, *error = NULL;
     gchar *error_message;
 
-    if(json_root != NULL) {
+    if(json_root) {
         json_obj = matrix_json_node_get_object(json_root);
-        errcode = matrix_json_object_get_string_member(json_obj, "errcode");
-        error = matrix_json_object_get_string_member(json_obj, "error");
+        if (json_obj) {
+            errcode = matrix_json_object_get_string_member(json_obj, "errcode");
+            error = matrix_json_object_get_string_member(json_obj, "error");
+        }
     }
 
-    if(errcode != NULL && error != NULL) {
+    if(errcode && error) {
         error_message = g_strdup_printf("%s: %s: %s",
                 _("Error from home server"), errcode, error);
     } else {
         error_message = g_strdup_printf("%s: %i",
                 _("Error from home server"), http_response_code);
     }
 
-    purple_connection_error_reason(ma->pc,
-            PURPLE_CONNECTION_ERROR_OTHER_ERROR,
-            error_message);
+    purple_notify_error(ma->pc, "Error", "Error", error_message);
```

Comment:
I'll move it to a new PR but basically why do you want to disconnect from the matrix server completely if the api returned a bad response? surely you just pop up an error message to the user and recover from there...